core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksStore.java

```diff
@@ -84,10 +90,29 @@ public RocksStore(String name, String dbPath, String checkpointPath,
       DBOptions dbOpts,
       Collection<ColumnFamilyDescriptor> columnFamilyDescriptors,
       List<AtomicReference<ColumnFamilyHandle>> columnHandles) {
+    this(name, dbPath, checkpointPath, null, null, dbOpts, columnFamilyDescriptors, columnHandles);
+  }
+
+  /**
+   * @param name a name to distinguish what store this is
+   * @param dbPath a path for the rocks database
+   * @param checkpointPath a path for taking database checkpoints
+   * @param parallelBackupPath a path for taking database backup in parallel
+   * @param parallelBackupPoolSize the thread pool size for taking database backup in parallel
+   * @param dbOpts the configured RocksDB options
+   * @param columnFamilyDescriptors columns to create within the rocks database
+   * @param columnHandles column handle references to populate
+   */
+  public RocksStore(String name, String dbPath, String checkpointPath, String parallelBackupPath,
```

Comment:
I think the extra `parallelBackupPath` is redundant, if `MASTER_PARALLEL_BACKUP_ROCKSDB=true` the backup path will be used to generate a backup in parallel.

Reply:
Is the parallel backup file different from the one you generate with single thread?

Reply:
parallelBackupPath needs to exist, because rocks is not only used by inodeStore, but also by blockMetaStore. Although I did not add parallel backup function to blockMetaStore. I can add to blockMetaStore if needed.

> Is the parallel backup file different from the one you generate with single thread?

There are some differences, you can see the comments about SeekableByteChannel in ZipArchiveOutputStream, but this difference has little effect in the alluxio scenario

Reply:
> Is the parallel backup file different from the one you generate with single thread?

The comment for ParallelScatterZipCreator mentions that there was a problem with the previous version, but that problem has been fixed.
I think if there is any other influence it will be mentioned in the comments of ParallelScatterZipCreator.

> There are some differences, you can see the comments about SeekableByteChannel in ZipArchiveOutputStream, but this difference has little effect in the alluxio scenario

I answered wrong, this is the effect of different output streams on ZipArchiveOutputStream

Reply:
okay let's do 3 things if possible:
1. Seems parallel compression can be applied to both inode store and block store. So if we use parallel compression, let's compress both with multiple threads. You can do it one by one, like create thread pool -> compress inode store -> compress block store -> close thread pool.
2. Ideally if you compress a file with single thread, you should be able to decompress that with multiple threads. Imagine you have a backup then changed the config, now after you start the master you should be able to read that backup. And vice versa. Is that possible? If a single-thread-compressed file can be decompressed by multiple threads, could you also add a unit test for that?
3. If a single-thread-compressed file can NOT be decompressed by multiple threads, you can use the file name to tell if the backup is generated by multiple threads. Like `alluxio-metadata-multithread.20220915.zip` vs `alluxio-metadata.20220915.zip` so you know if multiple threads should be used to decompress.

Reply:
Is there additional overhead of using parallel compression/decompression? It looks like it may need to make an extra copy of the data, is that correct?

I would think that we could just have one code path, so we don't have two types of compression options, so we can remove the old sequential compression, then if people want sequential compression they just use a single thread.

Reply:
@tcrain The earliest implementation was somewhat different from the current implementation. Decompression now requires copying the data from the input stream to a local file, and then reading the file in multiple threads to decompress.

Both implementations should to be reserved now. Because the old implementation is targz and the new implementation is zip, the two are not interchangeable.